### PR TITLE
Form Migrations: Button

### PR DESCRIPTION
**Pull request by tskarhed**
Seemed to work relatively painless. #22862

Still need to handle `.btn` styles for backwards compability.

### CODE REVIEW

**Review by torkelo — APPROVED** [review]
Looks good!

File: packages/grafana-ui/src/components/ClipboardButton/ClipboardButton.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import React, { PureComponent } from 'react';
 import Clipboard from 'clipboard';
-import { Button, ButtonProps } from '../Button/Button';
+import { Button, ButtonProps } from '../Forms/Button';
```
Comment (torkelo) [inline packages/grafana-ui/src/components/ClipboardButton/ClipboardButton.tsx]: This change looks strange, should it not be that we move the old Button to /Forms/Legacy ?

**Review by torkelo — COMMENTED** [review]
Is not better to move the new Button the the out of the Forms namespace at the same time? (So remove current Button and the one under Forms/Button to the component namespace

Otherwise, you need to touch all these files again.

**Comment by torkelo** [comment]
I do not think you need to modify the btn styles right now, important point is to make grafana/ui ready for beta (so components in the right place and the component props like we want them)

File: packages/grafana-ui/src/components/CallToActionCard/CallToActionCard.story.tsx [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { storiesOf } from '@storybook/react';
 import { renderComponentWithTheme } from '../../utils/storybook/withTheme';
 import { CallToActionCard } from './CallToActionCard';
 import { select, text } from '@storybook/addon-knobs';
-import { Button } from '../Button/Button';
+import { Button } from '../Forms/Legacy/Button/Button';
```
Comment (torkelo) [inline packages/grafana-ui/src/components/CallToActionCard/CallToActionCard.story.tsx]: Do we need the legacy Button? what are the api differences?

Comment (tskarhed) [reply packages/grafana-ui/src/components/CallToActionCard/CallToActionCard.story.tsx]: No, I don't think so. The only difference is is variants. Currently looking at how much we can move into the new Button.

Comment (tskarhed) [reply packages/grafana-ui/src/components/CallToActionCard/CallToActionCard.story.tsx]: There are components like the FullWidthButtonContainer that needs to be taken into consideration too.

**Comment by tskarhed** [comment]
In the DatasourceHttpSettings comonent
Screenshot from 2020-03-25 14-21-58
LegacyValueMappingsEditor (which is going to be gone by 7.0, so it doesn't matter?)
Screenshot from 2020-03-25 14-25-56

**Comment by torkelo** [comment]
think those are fine, we will likley update those forms in time

File: packages/grafana-ui/src/components/Layout/Layout.story.tsx [hunk 3]
```diff
@@ -11,7 +11,7 @@ export default {
   decorators: [withStoryContainer, withCenteredStory, withHorizontallyCenteredStory],
 };
 
-const justifyVariants = ['flex-start', 'flex-end', 'space-between'];
+const justifyVariants = ['flex-start', 'flex-end', 'space-betw  een'];
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/Layout/Layout.story.tsx]: Oopsie

File: packages/grafana-ui/src/components/ValueMappingsEditor/ValueMappingsEditor.tsx [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import { MappingType, ValueMapping } from '@grafana/data';
-import Forms from '../Forms';
+import { Button } from '../Button/Button';
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/ValueMappingsEditor/ValueMappingsEditor.tsx]: Can shorten this one

File: packages/grafana-ui/src/components/index.ts [hunk 5]
```diff
@@ -99,7 +98,7 @@ export { LogLabels } from './Logs/LogLabels';
 export { LogRows } from './Logs/LogRows';
 export { getLogRowStyles } from './Logs/getLogRowStyles';
 export { ToggleButtonGroup, ToggleButton } from './ToggleButtonGroup/ToggleButtonGroup';
-// Panel editors
+// Panel editors./Forms/Legacy/Button/FullWidthButtonContainer
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/index.ts]: Things auto import fixer does

File: packages/grafana-ui/src/components/Button/Button.mdx [hunk 6]
```diff
@@ -59,5 +59,22 @@ Used for removing or deleting entities.
   </div>
 </Preview>
 
+## Link
+
+Used for for hyperlinks.
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/Button/Button.mdx]: for for for?

**Comment by torkelo** [comment]
Dashboard settings: 

image

File: packages/grafana-ui/src/components/Button/Button.tsx [hunk 7]
```diff
@@ -1,72 +1,182 @@
...
+
+  return {
+    button: cx(
+      css`
+        label: button;
+        display: inline-flex;
+        align-items: center;
+        font-weight: ${theme.typography.weight.semibold};
+        font-family: ${theme.typography.fontFamily.sansSerif};
+        line-height: ${theme.typography.lineHeight.sm};
```
Comment (torkelo) [inline packages/grafana-ui/src/components/Button/Button.tsx]: remove the line-height, should not be here

**Comment by torkelo** [comment]
pushed an update to align the primary button color for old btn-primary

**Comment by tskarhed** [comment]
Still exporting Button and LinkButton under `Forms`, but this should be removed after https://github.com/grafana/grafana-enterprise/pull/485 has been merged

**Review by torkelo — APPROVED** [review]
Looks good now!

**Pull request merged**